### ISSUE TITLE
Update $rootScope before controller declaration

### ISSUE THREAD

**Issue by Machiaweliczny** [angular/angular.js#12827]
Hey I've code after compilation of coffescript like this:
```
(function() {
  var rootScope;

  rootScope = angular.element(document).scope();

  if (rootScope) {
    rootScope.sourcePath = 'app/assets/javascripts/main/controllers/edit_profile_controller.coffee';
  }

  angular.module('Main').controller('EditProfileCtrl', ... something
```

I want to add this sourcePath variable to every directive or controller $scope. The approach above that I tried doesn't work but demonstates what I want to achieve and I would love if you could help me with this:

Given that before controller declaration I've some variable in it's closure - append this variable to controller scope.

How can I achieve it?

**Comment by Narretz** [comment]
You should assign the variable to the rootscope inside your module's run function. Then it'll be available in all controllers / scopes. Except isolate scopes.

**Comment by Machiaweliczny** [comment]
The point is I want to have it different almost for every file. I'm not sure if you are familiar how coffescript files are preprocesed and merged into one, but basically it's like this:
I have `/controller/foo` and `/controllers/bar` and its will end up like this after compilation:
```
(function(){
  var sourcePath = "/controllers/foo.coffee"
  angular.module("x").conroller("foo", ...
})
(function(){
 var sourcePath = "/conrollers/bar.coffee"
 angular.module("x").controller("bar", ...
)
```
So I've to change it before each declaration in module run it's not enough. I dediced it would be too hacky to achieve it with javascript, because as you mentioned even if I updated $rootScope it wouldn't add this to directives with isolated ones, so I basically wrote regex that detects controllers definitions and directives link / controller functions and I simply append this `sourcePath` to declaration and it does the job. It's easy to incorporate it in any assets pipeline, so I'm satisfied without hacky JS solution. Just tiny bit sad, because I love hacky solutions ;)
